core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -116,9 +120,12 @@ private[spark] class ExecutorAllocationManager(
   // TODO: The default value of 1 for spark.executor.cores works right now because dynamic
   // allocation is only supported for YARN and the default number of cores per executor in YARN is
   // 1, but it might need to be attained differently for different cluster managers
-  private val tasksPerExecutor =
+  private val tasksPerExecutorForFullParallelism =
```

Comment:
We don't really need this variable now, can we just remove it?

Reply:
I was originally thinking we may avoid introducing the concept `tasksPerExecutorForFullParallelism`, but rather only have executorCores and taskCPUs, but I don't have a strong opinion over that.

Reply:
This is not exposed, it is merely a more precise description of the actual computation. I just wanted to state more clearly that the existing default behavior is maximizing the parallelism

Reply:
ok